test/locks.js

```diff
@@ -39,231 +47,177 @@ describe('Locks', function() {
-        // listen normally and not try to run a task. -Tom
-        apos.argv._ = [];
-        return callback(null);
-      },
-      afterListen: function(err) {
-        assert(!err);
-        done();
       }
     });
+
+    assert(apos.modules['apostrophe-locks']);
+    assert(apos.modules['apostrophe-locks-1']);
+    assert(apos.modules['apostrophe-locks-2']);
+    assert(apos.modules['apostrophe-locks-3']);
   });
 
-  it('cleanup', function(done) {
-    apos.locks.db.remove({}, function(err) {
-      assert(!err);
-      done();
-    });
+  it('cleanup', async function() {
+    await apos.locks.db.remove({});
   });
 
-  it('should allow a single lock without contention uneventfully', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock, unlock ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock(callback) {
-      return locks.lock('test', callback);
-    }
-    function unlock(callback) {
-      return locks.unlock('test', callback);
-    }
+  it('should allow a single lock without contention uneventfully', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    await locks.lock('test');
+    await locks.unlock('test');
   });
 
-  it('should allow two differently-named locks uneventfully', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock1, lock2, unlock1, unlock2 ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock1(callback) {
-      return locks.lock('test1', callback);
-    }
-    function unlock1(callback) {
-      return locks.unlock('test1', callback);
-    }
-    function lock2(callback) {
-      return locks.lock('test2', callback);
-    }
-    function unlock2(callback) {
-      return locks.unlock('test2', callback);
-    }
+  it('should allow two differently-named locks uneventfully', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    await locks.lock('test1');
+    await locks.lock('test2');
+    await locks.unlock('test1');
+    await locks.unlock('test2');
   });
 
-  it('should flunk a second lock by the same module', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock, lockAgain, unlock, unlockAgain ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock(callback) {
-      return locks.lock('test', callback);
-    }
-    function lockAgain(callback) {
-      return locks.lock('test', function(err) {
-        // SHOULD fail
-        assert(err);
-        return callback(null);
-      });
-    }
-    function unlock(callback) {
-      return locks.unlock('test', callback);
+  it('should flunk a second lock by the same module', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    await locks.lock('test');
+
+    try {
+      await locks.lock('test');
+      assert(false);
+    } catch (e) {
+      assert(e);
```

Comment:
This is a nice way to show intent although it is not strictly necessary to have this assert, since the only way assert(false) can happen is if the desired exception is not thrown.